changelog: Add placeholder directory to add the changelogs (#250)
flatcar-linux/portage-stable: changelog/README.md

The `changelog` directory contains the description of the changes introduced into the repository.

> The changes are essentially divided into 4 categories

Reviewer: ```suggestion
The changes are essentially divided into 4 categories:
```

> - changes: List of PRs brings Changes and/or Enhancements
> - bugfixes: List of PRs fixing existing issues
> - security: List of PRs fixing security issues
> - updates: Package Updates

Reviewer: ```suggestion
- changes: PRs bringing Changes and/or Enhancements
- bugfixes: PRs fixing existing issues
- security: PRs fixing security issues
- updates: PRs updating packages
```

## How to add the file

Based on the category the PR falls into create a new file in the respective directory

> with the filename format `DD-MM-YYYY-<few-words-about-the-change>.md`

Reviewer: We can suggest a help to generate the date format:

```suggestion
with the filename format `DD-MM-YYYY-<few-words-about-the-change>.md (can be generated via: `$(date '+%d-%m-%Y')-<few-words-about-the-change>.md``
```

> The contents of the file should describe the changes in an ellaborative manner, except for `Updates`.

Reviewer: We could maybe add an entry example:

```md
$ cat `changelog/bugfixes/(date '+%d-%m-%Y')-selinux-torcx.md
add SELinux rule to allow torcx to correctly start in SELinux enforced mode ([#1426](https://github.com/flatcar-linux/coreos-overlay/pull/1426))
```
Since we give an `Update` example right after, that would make sense. :)

Reviewer 2: Do you also have an example how to use flog to generate an update entry with a link to the changelog or an CVE list entry? Or is this not ready yet for this use case?

Reviewer 3: ```suggestion
The contents of the file should describe the changes in an elaborative manner, except for `Updates`.
```

Author: I did type the correct spelling of elaborative, not sure why vim suggested me double-l 😖

Author: The intention of `flog` is different, it would be a central place to all the project changelog. My idea is to have a script to do the release, something like [blurb](https://pypi.org/project/blurb/) but for us, it would be just a small script.

> The contents of the file should describe the changes in an elaborative manner, except for `Updates`.

Reviewer 2: ```suggestion
The contents of the file should describe the changes in an elaborative manner (use the past tense for the change/bugfix description to avoid confusion with the imperative voice for actions the user should do as a result).
Security fixes of upstream packages and package updates can be kept short in most cases and follow a standard format.
```

> As `Updates` refer to the pacakage updates, the description of the file should be of the

Reviewer: ```suggestion
As `Updates` refer to the package updates, the description of the file should be of the
```

> following format: `Package Name [Version](link to changelog)`

Reviewer 3: I would like to see an example of it. I get the `Package Name` part. It would be something like `dev-util/perf`. `Version` would be version taken from ebuild filename, right (so `5.15-r1`)? I don't get the `link to changelog` part - what changelog?

Reviewer 2: It means the release notes of the updated package

Reviewer 3: Could be tricky for some projects. What would use for `dev-util/perf`? https://kernelnewbies.org/Linux_5.15#Tracing.2C_perf_and_BPF for example?

Reviewer 4: Package name is the upstream name, no need to prefix the group. Also, version is the upstream release version w/o recipe suffix. Changelog is a link to the upstream release changelog (if available).
A good example would be previously updated item from an existing changelog, e.g. Linux in the latest stable release:

```
# Updates
- Linux ([5.10.77](https://lwn.net/Articles/874852/))
```

Reviewer 3: Cool, thanks for the example. I think it needs to be clarified in the README.
